### DISCUSSION TITLE
Failed to refresh device state

### DISCUSSION THREAD

**Discussion by demonkazuya** [make-all/tuya-local#554] [Q&A]
hi

since I added more devices, I keep on getting the error. Whenever I try reboot home assistant a random device will be working again, and a random device will get the error. I'm thinking it's tied to the number of sensors sending information. This happened when I added the smart plugs after the last integration update. here is a screenshot of the log that I'm having:

![image](https://user-images.githubusercontent.com/21069535/226184258-8231e59e-143e-4015-a95a-24c98832a588.png)

**Comment by make-all** [comment]
If you have any devices with the protocol set to "auto", try to find out what protocol version they are using, and manually set it to that.
The "auto" protocol detection is just a trial and error process, and some devices stop responding for some time after an error, so often when the correct protocol version is tried, the device is not responding and that chance to start working is missed.

**Reply by demonkazuya** [reply]
do I need to remove all devices and add it back again using the correct protocol version or just the once that I'm having issues with? because right now after re-adding one device, I'm still getting the error

**Reply by demonkazuya** [reply]
after removing all the devices that has refresh error, I'm not able to add any devices any more even if I hard code all the fields (device id, IP, local key, protocol) even if I do auto on protocol, it just gives me this error:

 ![image](https://user-images.githubusercontent.com/21069535/226188909-3d2417d6-de14-4ffd-8385-d7534283e213.png)

I tried restarting home assistant a couple of times but I'm still unable to add devices. I don't see any error message or logs related to adding the device. I'm starting to think that there is something limiting the number of devices that can be added.

**Reply by make-all** [reply]
You didn't need to remove the devices, there is a Configure button in the Devices & Services Integrations page for each device that allows reconfiguring the connection settings. How did you remove them? If you just removed the entities the device will still be there preventing the duplicates from being registered. Otherwise, there is another topic on connection issues that has some things to check for.

**Reply by demonkazuya** [reply]
this how I delete the devices:

![image](https://user-images.githubusercontent.com/21069535/226286718-6c1c4088-0980-4d67-82c2-7185d2385fde.png)

![image](https://user-images.githubusercontent.com/21069535/226286791-53a83002-e057-4674-a0ac-7439273a1376.png)

did some experimentation, I removed 1 device from the integration and tried to add a different device and it worked. Somehow I think im limited to the number of devices that I can add

BTW all the devices currently added has the protocol hardcoded no auto

**Reply by Casadomoticap** [reply]
I have the same problem and I opened this discussion: https://github.com/make-all/tuya-local/discussions/553#discussioncomment-5360055 
So I think that there are some code in the project that make this error or something like that

**Comment by make-all** [comment]
You may be running into an open socket limit imposed by Python, or Home Assistant.  Try turning on polling for some of your devices that you do not care so much about real-time updates from.

**Reply by demonkazuya** [reply]
I tried polling for devices such as light switch and I'm able to add more devices. However, I think I'm still being limited because after adding like 3 more devices I get the same issue. Is the limit related to the number of entities? I'm thinking to disable some entities that is not important such as voltage and current reading on my power metered devices.

**Comment by dmshimself** [comment]
Is this error always an error?  If my light switches are physically turned off and thus have no power, my HA log fills quite rapidly with these error messages until the lights are turned on.  Should the message be a warning rather than an error?
